src/win/conpty.cc

```diff
@@ -8,6 +8,12 @@
  *   with pseudo-terminal file descriptors.
  */
 
+// node versions lower than 10 define this as 0x502 which disables many of the definitions needed to compile
+#include <node_version.h>
+#if NODE_MODULE_VERSION <= 57
+  #define _WIN32_WINNT 0x600
```

Comment:
Can you move this condition below `#include <nan.h>` as it already includes `node_version` and also defines `NODE_8_0_MODULE_VERSION`

So essentially this could be
```
diff --git a/src/win/conpty.cc b/src/win/conpty.cc
index 53e8caf..de67282 100644
--- a/src/win/conpty.cc
+++ b/src/win/conpty.cc
@@ -18,6 +18,11 @@
 #include <strsafe.h>
 #include "path_util.h"

+#if NODE_MODULE_VERSION <= NODE_8_0_MODULE_VERSION
+#undef _WIN32_WINNT
+#define _WIN32_WINNT 0x600
+#endif
+
 extern "C" void init(v8::Local<v8::Object>);

 // Taken from the RS5 Windows SDK, but redefined here in case we're targeting <= 17134
```

Reply:
I can't move it lower because `#include <nan.h>` is what pulls in the windows function definitions.

Reply:
Maybe its worth fixing in `NAN` then.

Reply:
I think it makes more sense here since this is safe due to the fact that the conpty code is only called on versions of windows that support all these APIs, node 8 can _technically_ run on windows XP which won't have these function definitions.